apps/wrapper/src/renderer/src/components/Fileclass.svelte

```diff
@@ -0,0 +1,161 @@
+<script lang="ts">
+  import logo from '../assets/logo.png';
+  import { onMount } from 'svelte';
+  import { appState } from '../stores';
+
+  let fileClass = '';
+  let appName = '';
+  let proceed = false;
+  let checkLoading = false;
+
+  onMount(async () => {
+    await window.api.getFileClass('3be2565e-fe51-4e9d-820f-d6c1315d204d');
+    let classInterval = setInterval(async () => {
+      const fileClassResponse = await window.api.checkFileClass();
+      if (fileClassResponse.code === 200) {
+        fileClass = fileClassResponse.message;
+        clearInterval(classInterval);
+      }
+    }, 1000);
+    const appNameRes = await window.api.getAppName();
+    appName = appNameRes.message;
+  });
+
+  async function topSecretChecks() {
+    if (fileClass === 'TOPSECRET') {
+      checkLoading = true;
+      const response: { code: number; message: string } = await window.api.secretChecks();
+      proceed = true;
+      console.log(response.code);
+      if (response.code === 200) {
+        appState.set({
+          passedCheck: true,
+          currentState: 'fileClass',
+          pingFailed: false,
+        });
+      } else {
+        proceed = false;
+        appState.set({
+          passedCheck: false,
+          currentState: 'fileClass',
+          pingFailed: false,
+        });
+      }
+    } else if (fileClass === 'SENSITIVE') {
+      proceed = true;
+      appState.set({
+        passedCheck: true,
+        currentState: 'fileClass',
+        pingFailed: false,
+      });
+    } else if (fileClass === 'OPEN') {
+      proceed = true;
+      appState.set({
+        passedCheck: true,
+        currentState: 'viewDoc',
+        pingFailed: false,
+      });
```

Comment:
Switch to using `.update()` instead of `.set()`

Reply:
Changed to
```
if (response.code === 200) {
        appState.update((state) => ({
          ...state,
          currentState: 'fileClass',
        }))
      } else {
        proceed = false;
        appState.update((state) => ({
          ...state,
          passedCheck: false,
        }))
      }
    } else if (fileClass === 'SENSITIVE') {
      proceed = true;
      appState.update((state) => ({
        ...state,
        currentState: 'fileClass',
      }));

    } else if (fileClass === 'OPEN') {
      proceed = true;
      appState.update((state) => ({
        ...state,
        currentState: 'viewDoc',
      }));
    }
```